src/backend/TrafficCourtsApi/TrafficCourts.Common.Test/Configuration/SplunkEventCollectorTest.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Moq;
+using NUnit.Framework;
+using Serilog;
+using Serilog.Events;
+using Serilog.Sinks.TestCorrelator;
+using TrafficCourts.Common.Configuration;
+
+namespace TrafficCourts.Common.Test.Configuration
+{
+    [TestFixture]
+    public class SplunkEventCollectorTest
+    {
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            Log.Logger = new LoggerConfiguration()
+                .WriteTo.TestCorrelator()
+                .CreateLogger();
+        }
+        
+        [Test]
+        public void Configure_checks_for_null_arguments()
+        {
+            var hostBuilderContext = new HostBuilderContext(new Dictionary<object, object>());
+            var loggerConfiguration = new LoggerConfiguration();
+
+            Assert.Throws<ArgumentNullException>(() => SplunkEventCollector.Configure(null, loggerConfiguration));
+            Assert.Throws<ArgumentNullException>(() => SplunkEventCollector.Configure(hostBuilderContext, null));
+        }
+        
+        [Test]
+        public void Configure_logs_warning_if_splunk_configuration_settings_not_found()
+        {
+            var configuration = new ConfigurationBuilder()
+                .Build();
+
+            var hostBuilderContext = new HostBuilderContext(new Dictionary<object, object>());
+            hostBuilderContext.Configuration = configuration;
+
+            var loggerConfiguration = new LoggerConfiguration();
+
+            using (TestCorrelator.CreateContext())
+            {
+                // act
+                SplunkEventCollector.Configure(hostBuilderContext, loggerConfiguration);
+
+                var logEvent = TestCorrelator.GetLogEventsFromCurrentContext().Single();
+
+                Assert.AreEqual(LogEventLevel.Warning, logEvent.Level);
+
+                Assert.NotNull(logEvent.RenderMessage()); // assert anything about this message?
+            }
+        }
+        
+        [Test]
+        [Ignore("Having troubles with Serilog reading the configuration and also our reading of the configuration")]
+        public void Configure_will_configure_EventCollector_if_url_and_token_are_available()
+        {
+            var configurationValues = new Mock<IDictionary<string, string>>();

```

Comment:
This test was skipped. I don't know how to get it to run.

![image](https://user-images.githubusercontent.com/5287074/106197340-d2c01580-616f-11eb-95c4-2eb6529725a2.png)